pyro/infer/mcmc/util.py

```diff
@@ -170,28 +170,30 @@ def _populate_cache(self, model_trace):
             raise ValueError("Finite value required for `max_plate_nesting` when model "
                              "has discrete (enumerable) sites.")
         model_trace.compute_log_prob()
+        model_trace.pack_tensors()
         for name, site in model_trace.nodes.items():
             if site["type"] == "sample" and not isinstance(site["fn"], _Subsample):
                 if is_validation_enabled():
                     check_site_shape(site, self.max_plate_nesting)
-                self.ordering[name] = frozenset(f for f in site["cond_indep_stack"] if f.vectorized)
-                log_prob_shape = site["log_prob"].shape
-                self._enum_dims[site["name"]] = set((i for i in range(-len(log_prob_shape), -self.max_plate_nesting)
-                                                     if log_prob_shape[i] > 1))
+                self.ordering[name] = frozenset(model_trace.plate_to_symbol[f.name]
+                                                for f in site["cond_indep_stack"]
+                                                if f.vectorized)
+        self._enum_dims = set(model_trace.symbol_to_dim) - set(model_trace.plate_to_symbol.values())
```

Comment:
+1 on the simplification, we only need per site enum dims for `TraceTreeEvaluator`.